src/store/recentConnectionAppsAtom.tsx

```diff
@@ -0,0 +1,34 @@
+import { atomWithStorage } from 'jotai/utils';
+import { atom } from 'jotai';
+import { SignClientTypes } from '@walletconnect/types';
+
+type MetadataWithDate = SignClientTypes.Metadata & {
+  date: Date;
+};
+
+type StoredValueType = Record<string, MetadataWithDate>;
+
+export const initialValue = {}
+
+export const recentConnectionAppsAtom = atomWithStorage<StoredValueType>(
+  "connectionApps",
+  initialValue,
+  undefined,
+  { getOnInit: true }
+);
+
+export const sortedRecentConnectionAppsAtom = atom(
+  (get): MetadataWithDate[] => {
+    const storedValue: StoredValueType = get(recentConnectionAppsAtom);
```

Comment:
You shouldn't need to type anything explicitly here, at least it worked for me locally
```suggestion
  (get) => {
    const storedValue = get(recentConnectionAppsAtom);
```